src/v2/content_digest.rs

```diff
@@ -0,0 +1,136 @@
+/// Implements types and methods for content verification
+use sha2::{self, Digest};
+use v2::*;
+
+/// ContentDigest stores a digest and its DigestAlgorithm
+#[derive(Clone, Debug, PartialEq)]
+pub struct ContentDigest {
+    digest: String,
+    algorithm: DigestAlgorithm,
+}
+
+/// DigestAlgorithm declares the supported algorithms
+#[derive(Display, Clone, Debug, PartialEq, EnumString)]
+pub enum DigestAlgorithm {
+    #[strum(to_string = "sha256")]
+    Sha256,
+}
+
+impl ContentDigest {
+    /// try_new attempts to parse the digest string and create a ContentDigest instance from it.error_chain
+    ///
+    /// Success depends on
+    /// - the string having a "algorithm:" prefix
+    /// - the algorithm being supported by DigestAlgorithm
+    pub fn try_new(digest: String) -> Result<Self> {
+        let digest_split = digest.split(':').collect::<Vec<&str>>();
+
+        if digest_split.len() != 2 {
+            return Err(format!("digest '{}' does not have an algorithm prefix", digest).into());
+        }
+
+        let algorithm =
+            std::str::FromStr::from_str(digest_split[0]).map_err(|e| format!("{}", e))?;
+        Ok(ContentDigest {
+            digest: digest_split[1..].join(""),
```

Comment:
I'm not sure I follow the `[1..].join("")` part. We checked above that we have exactly two fields, so I would expect a `[1]` here, no?